test/distributed/org/apache/cassandra/distributed/test/sai/IndexStreamingFailureTest.java

```diff
@@ -0,0 +1,200 @@
+
+        IInvokableInstance first = cluster.get(1);
+        IInvokableInstance second = cluster.get(2);
+        first.runOnInstance(()-> DatabaseDescriptor.setStreamEntireSSTables(streamEntireSSTables));
+        second.runOnInstance(()-> DatabaseDescriptor.setStreamEntireSSTables(streamEntireSSTables));
+
+        first.executeInternal(String.format("INSERT INTO %s.%s(pk, v) VALUES (?, ?)", KEYSPACE, table), 1, "v1");
+        first.flush(KEYSPACE);
+
+        Object[][] rs = second.executeInternal(String.format("select pk from %s.%s where v = ?", KEYSPACE, table), "v1");
+        assertThat(rs.length).isEqualTo(0);
+
+        // The repair job should fail when index completion fails. This should also fail the streaming transaction.
+        NodeToolResult result = second.nodetoolResult("repair", KEYSPACE);
+        result.asserts().failure().errorContains(TEST_ERROR_MESSAGE);
+
+        // The SSTable should not be added to the table view, as the streaming transaction failed...
+        rs = second.executeInternal(String.format("select pk from %s.%s where pk = ?", KEYSPACE, table), 1);
+        assertThat(rs.length).isEqualTo(0);
+
+        // ...and querying the index also returns nothing, as the index for the streamed SSTable was never built.
+        rs = second.executeInternal(String.format("select pk from %s.%s where v = ?", KEYSPACE, table), "v1");
+        assertThat(rs.length).isEqualTo(0);
+
+        // On restart, ensure that the index remains querable and does not include the data we attempted to stream. 
+        second.shutdown().get();
+        second.startup();
+
+        // On restart, the base table should be unchanged...
+        rs = second.executeInternal(String.format("select pk from %s.%s where pk = ?", KEYSPACE, table), 1);
+        assertThat(rs.length).isEqualTo(0);
+
+        // ...and the index should remain queryable, because from its perspective, the streaming never happened.
+        rs = second.executeInternal(String.format("select pk from %s.%s where v = ?", KEYSPACE, table), "v1");
+        assertThat(rs.length).isEqualTo(0);
+
+        // Disable failure injection, and verify that the index is queryable and has the newly streamed data:
+        second.runOnInstance(()-> ByteBuddyHelper.failFlush = false);
+        second.runOnInstance(()-> ByteBuddyHelper.failValidation = false);
+        second.nodetoolResult("repair", KEYSPACE).asserts().success();
+
+        rs = second.executeInternal(String.format("select pk from %s.%s where v = ?", KEYSPACE, table), "v1");
+        assertThat(rs.length).isEqualTo(1);
+    }
+
+    public static class ByteBuddyHelper
+    {
+        volatile static boolean failFlush = false;
+        volatile static boolean failValidation = false;
+        
+        @SuppressWarnings("resource")
+        static void installErrors(ClassLoader loader, int node)
+        {
+            if (node == 2)
```

Comment:
The two `if` blocks can be merged, or we could just have `if (node != 2) return;` at the beginning of the method.

Reply:
Done